nodes/modifier_change/offset_line.py

```diff
@@ -189,27 +205,22 @@ def sv_init(self, context):
 
     def process(self):
 
-        if not (self.inputs['Vers'].is_linked or self.inputs['Edgs'].is_linked):
+        if not any(socket.is_linked for socket in self.inputs):
```

Comment:
```diff
- if not any(socket.is_linked for socket in self.inputs):
+ if not all(socket.is_linked for socket in self.inputs[:2]):
```
the first two sockets must be connected, last socket is optional.

Reply:
there's differing opinions about whether we need these input checks, in my opininion they are fine. but we also have a mechanism that responds to sockets that are inconnected and don't have a "default" parameter in the `socket.sv_get(   )` call. -- in that scenario the node will say " socket x not connected " [(or something similar)](https://github.com/nortikin/sverchok/pull/1893)

Reply:
@portnov  in the case of two sockets been mandatory, we could maybe raise a specific error too ? like
(a proposal, this won't work at the moment)
```python
   raise SvNotFullyConnected(self, msg="node x needs input a and input b to perform work")
```

Reply:
@zeffii good idea. I'm all for more specific exception classes for specific cases (for now we raise plain Exception in many cases).

Reply:
@zeffii 
> but we also have a mechanism that responds to sockets that are unconnected and don't have a "default" parameter in the `socket.sv_get( )`

So will it be better to delete checking at all?

Reply:
for now use 
```python
if not all(socket.is_linked for socket in self.inputs[:2]):
    return
```
we will add the correct Exception class soon!